pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -76,17 +78,39 @@ func newClientWithWatch(kubeconfig []byte, options client.Options) (client.WithW
 	return client.NewWithWatch(restConfig, options)
 }
 
+type multiKueueWatcher interface {
+	// returns an empty list of objets
+	GetEmptyList() client.ObjectList
+	// returns the key of the workload of interest
+	// - the object name for workloads
```

Comment:
I think for workloads it fails looking at the impl?

Reply:
Oh, sorry, I got confused by the same function implemented both for Workloads and Jobs. I looked at the Jobs implementations and thought it would error for Workloads. 

Now I see what you do, though the extra layer of indirection makes it slightly hard to follow, but I see there are trade-offs, and this way you can commonize the code for Workloads and Jobs.